Remove ui api non-repo collections endpoint

awcrosby opened the pull request:
This endpoint is no longer used by the UI and was replaced with per-repo collections logic.

newswangerd reviewed (changes requested):
This looks good, but it shouldn't be merged until after the next cloud release happens.

awcrosby commented:
@newswangerd I added back in the _ui/collections endpoint as deprecated and changed it to use the `published` repo so it can be removed after the UI deployment.

alikins commented on galaxy_ng/app/api/ui/viewsets/collection.py:
```diff
@@ -168,9 +54,19 @@ def get_queryset(self):
+            return serializers.CollectionDetailSerializer
+
+
+# TODO: Remove when ui is updated and no longer uses this endpoint
```
This is related to the cloud deploy stuff discussed yesterday?  ie, needing to have both versions of this endpoint running until there is no ui running that needs it?
